ui-manchette-with-spacetimechart/src/helpers.ts

```diff
@@ -0,0 +1,168 @@
+import type {
+  OperationalPoint,
+  SpaceTimeChartProps,
+} from '@osrd-project/ui-spacetimechart/dist/lib/types';
+
+import { BASE_OP_HEIGHT, MAX_TIME_WINDOW, MAX_ZOOM_X, MIN_ZOOM_X } from './consts';
+import type {
+  OperationalPointType,
+  ProjectPathTrainResult,
+  StyledOperationalPointType,
+} from '../types';
+import { getHeightWithoutLastWaypoint, msToS } from './utils';
+
+export const calcOperationalPointsToDisplay = (
+  operationalPoints: OperationalPointType[],
+  height: number,
+  isProportional: boolean,
+  zoom: number
+): StyledOperationalPointType[] => {
+  // For proportional display, we only display points that do not overlap with
+  // the last displayed point:
+  if (isProportional && operationalPoints.length > 0) {
+    // We start with the first operational point:
+    const result: StyledOperationalPointType[] = [
+      { ...operationalPoints[0], display: true, styles: {} },
+    ];
+    const from = operationalPoints.at(0)!.position;
+    const to = operationalPoints.at(-1)!.position;
+    const totalDistance = to - from;
+    const heightWithoutFinalOp = getHeightWithoutLastWaypoint(height);
+    let lastDisplayedOP = result[0];
+
+    // We iterate through all points, and only add them if they don't collide
+    // with the last visible point:
+    for (let i = 1, l = operationalPoints.length; i < l; i++) {
+      const op = operationalPoints[i];
+      const diff = op.position - lastDisplayedOP.position;
+      const display = (diff / totalDistance) * heightWithoutFinalOp * zoom >= BASE_OP_HEIGHT;
+      result.push({
+        ...op,
+        display,
+        styles: {},
+      });
+
+      if (display) lastDisplayedOP = result[i];
+    }
+
+    // In the end, to make sure the last point is visible, if it's not, we swap
+    // it with the last visible item:
+    const lastItem = result[result.length - 1];
+    if (!lastItem.display) {
+      const lastVisibleItem = result.findLast((op) => op.display);
+      if (lastVisibleItem) {
+        lastVisibleItem.display = false;
+        lastItem.display = true;
+      }
+    }
+
+    return result;
+  }
+  // For non-proportional display, we always display all the operational points:
+  else {
+    return operationalPoints.map((op) => ({ ...op, styles: {}, display: true }));
+  }
```

Comment:
```suggestion
    return operationalPoints.map((op) => ({ ...op, styles: {}, display: true }));
```

Reply:
i don't see the difference but maybe it's a doublon from the @kmer2016 commentary